qiling/extensions/pipe.py

```diff
@@ -84,3 +95,70 @@ def flush(self) -> None:
 
     def writable(self) -> bool:
         return True
+
+class SimpleBufferedStream(TextIO):
+    """Simple buffered IO.
+    """
+
+    def __init__(self):
+        self.buff = bytearray()
+        self.cur = 0
+
+    def lseek(self, offset: int, origin: int) -> int:
+        if origin == 0: # SEEK_SET
+            base = 0
+        elif origin == 1: # SEEK_CUR
+            base = self.cur
+        else: # SEEK_END
+            base = len(self.buff) - 1
+
+        if base + offset >= len(self.buff):
+            self.cur = base + offset - 1
+        else:
+            self.cur = base + offset
+
+        return self.cur
+
+    def seek(self, offset: int, origin: int) -> int:
+        return self.lseek(offset, origin)
+    
+    def tell(self) -> int:
+        return self.cur
+
+    def read(self, n: int = -1) -> bytes:
+        if n == -1:
+            ret = self.buff
+        else:
+            ret = self.buff[self.cur:self.cur + n]
+
+            if self.cur + n >= len(self.buff) - 1:
+                self.cur = len(self.buff)
+            else:
+                self.cur = self.cur + n
+
+        return bytes(ret)
+
+    def readline(self, limit: int = -1) -> bytes:
+        ret = bytearray()
+
+        while not (ret.endswith(b'\n') or len(ret) == limit):
+            ret.extend(self.read(1))
+
+        return bytes(ret)
+
+    def write(self, s: bytes) -> int:
+        self.buff.extend(s)
```

Comment:
Writing to a buffered stream should overwrite the content starting from current position.

Reply:
Done